### PR TITLE
INFO log for delete function; 204 status code on delete function

### LINKED ISSUE #228: API for Delete function should return 204
I feel since the Delete function does not return any content, shouldn't we return 204 (No Content), rather than 200 OK.

### LINKED ISSUE #230: API calls for Delete function is not logged when in -log-level=debug
When a function is registered or updated, it is logged, but not in the case of when a function is deleted. Is there is a specific reason for it?

```
INFO	Running in development mode with embedded etcd.
DEBUG	Function registered.	{"functionId": "getUser", "type": "awslambda"}
DEBUG	Function cache received set key.	{"key": "getUser", "value": "{\"functionId\":\"getUser\",\"provider\":{\"type\":\"awslambda\",\"arn\":\"arn:aws:lambda:us-east-1:063733696545:function:user-service-prod-getUser\",\"region\":\"us-east-1\"}}"}
DEBUG	Function registered.	{"functionId": "test", "type": "awslambda"}
DEBUG	Function cache received set key.	{"key": "test", "value": "{\"functionId\":\"test\",\"provider\":{\"type\":\"awslambda\",\"arn\":\"arn:aws:lambda:us-east-1:063733696545:function:user-service-prod-test\",\"region\":\"us-east-1\"}}"}
DEBUG	Function registered.	{"functionId": "test2", "type": "awslambda"}
DEBUG	Function cache received set key.	{"key": "test2", "value": "{\"functionId\":\"test2\",\"provider\":{\"type\":\"awslambda\",\"arn\":\"arn:aws:lambda:us-east-1:063733696545:function:user-service-prod-test2\",\"region\":\"us-east-1\"}}"}
DEBUG	Function updated.	{"functionId": "getUser"}
DEBUG	Function cache received set key.	{"key": "getUser", "value": "{\"functionId\":\"getUser\",\"provider\":{\"type\":\"awslambda\",\"arn\":\"arn:aws:lambda:us-east-1:063733696545:function:user-service-prod-getUserUpdated\",\"region\":\"us-east-1\"}}"}
DEBUG	Function updated.	{"functionId": "getUser"}
DEBUG	Function cache received set key.	{"key": "getUser", "value": "{\"functionId\":\"getUser\",\"provider\":{\"type\":\"awslambda\",\"arn\":
```

### PR DESCRIPTION
**Pull request by alexdebrie**
Fixes #230 

Fixes #228

### CODE REVIEW

File: functions/functions.go [hunk 1]
```diff
@@ -114,6 +114,9 @@ func (f *Functions) DeleteFunction(id FunctionID) error {
 	if err != nil {
 		return &ErrorNotFound{id}
 	}
+
+	f.Log.Info("function deleted", zap.String("functionId", string(id)))
```
Comment (mthenw) [inline functions/functions.go]: "Function"

Comment (alexdebrie) [inline functions/functions.go]: @mthenw For the message (`Function deleted`) or for the name of the key in the key-value pair? This tracks the other logs

Comment (mthenw) [inline functions/functions.go]: For the message. Hmm I think that every other log message starts with capital letter. If not it has to be fixed.

Comment (alexdebrie) [inline functions/functions.go]: Alright, changed for that file at least. Didn't check other files.

**Comment by mthenw** [comment]
@alexdebrie looks like you are making this PR from not updated master. https://github.com/serverless/event-gateway/blob/master/functions/functions.go#L41

**Comment by mthenw** [comment]
Can you rebase?

**Comment by mthenw** [comment]
I fix that in https://github.com/serverless/event-gateway/pull/235. Nevertheless, thank you for the contribution!

**Pull request closed without merging**